datascience/tables.py

```diff
@@ -395,17 +395,156 @@ def append_column(self, label, values):
+        z      | 1     | 10     | 40
+        >>> table3
+        letter | count | points | new_col1 | new_col2
+        a      | 9     | 1      | 10       | hello
+        b      | 3     | 2      | 20       | hello
+        c      | 3     | 2      | 30       | hello
+        z      | 1     | 10     | 40       | hello
+        >>> table
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
+
+        >>> table.append_column(123, [1, 2, 3, 4])
+        <ValueError raised>
+        >>> table.append_column('bad_col', [1, 2])
+        <ValueError raised>
+        """
+        new_table = self.copy()
+        new_table.append_column(label, values)
+        return new_table
+
+
     def relabel(self, column_label, new_label):
-        """Change the label of a column."""
-        assert column_label in self._columns
-        rewrite = lambda s: new_label if s == column_label else s
+        """Change the labels of columns specified by ``column_label`` to
+        labels in ``new_label``.
+
+        Args:
+            ``column_label`` (single str or list/array of str): The label(s) of
+                columns to be changed. Must be str.
+
+            ``new_label`` (single str or list/array of): The new label(s) of
+                columns to be changed. Must be str.
+
+                Number of elements must match number of elements in
+                ``column_label``.
+
+        Returns:
+            None
+
+        >>> table = Table([(1, 2, 3), (12345, 123, 5123)], ['points', 'id'])
+        >>> table.relabel('id', 'yolo')
+        >>> table
+        points | yolo
+        1      | 12345
+        2      | 123
+        3      | 5123
+        >>> table.relabel(['points', 'yolo'], ['red', 'blue'])
+        >>> table
+        red    | blue
+        1      | 12345
+        2      | 123
+        3      | 5123
+        """
+        if isinstance(column_label, str) and isinstance(new_label, str):
+            column_label, new_label = [column_label], [new_label]
+        d = dict(zip(column_label, new_label)) # dictioanry to map old labels to new ones
+        for col in d.keys():
+            assert col in self._columns
+        rewrite = lambda s: d[s] if s in d.keys() else s
         columns = [(rewrite(s), c) for s, c in self._columns.items()]
         self._columns = collections.OrderedDict(columns)
-        if column_label in self._formats:
-            formatter = self._formats.pop(column_label)
-            self._formats[new_label] = formatter
+        for label in self._formats:
+            if label in column_label:
+                formatter = self._formats.pop(label)
+                self._formats[d[label]] = formatter
         return self
 
+    def with_relabeling(self, column_label, new_label):
+        """Returns a new table with column_label changed to new_label.
+        ``column_label`` and ``new_label`` may be single values or lists
```

Comment:
Let's insert a newline above the line starting with column_label and new_label so that the summary on the main table doc page is short.